e2e_playwright/st_dataframe_input_data_test.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+def test_dataframe_input_format_rendering(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that st.dataframe renders various data formats correctly via snapshot testing."""
+    dataframe_elements = app.get_by_test_id("stDataFrame")
+    expect(dataframe_elements).to_have_count(34)
+
+    # The dataframe component might require a bit more time for rendering the canvas
+    app.wait_for_timeout(250)
+
+    for i, element in enumerate(dataframe_elements.all()):
+        assert_snapshot(element, name=f"st_dataframe-input_data_{i}")
```

Comment:
Hmmm this one is a pretty large test and we just index it without adding descriptions. It seems like the other ones have descriptive names so can we add a TODO here to add descriptive names rather than indexing on `i`?

Reply:
Yep indeed, but that is kind of intended here since its 1) a lot of different formats 2) just iterates a list of test examples that we are also use in different types of tests (e.g. unit tests). So I want to make it super easy in that case to just add additional cases.